src/data/equiv/fin.lean

```diff
@@ -37,6 +37,26 @@ def fin_two_equiv : fin 2 ≃ bool :=
   end,
   assume b, match b with tt := rfl | ff := rfl end⟩
 
+/-- The 'identity' equivalence between `fin n` and `fin m` when `n = m`. -/
+def fin_congr {n m : ℕ} (h : n = m) : fin n ≃ fin m :=
+equiv.subtype_equiv_right (λ x, by subst h)
+
+@[simp] lemma fin_congr_apply_mk {n m : ℕ} (h : n = m) (k : ℕ) (w : k < n) :
+  fin_congr h ⟨k, w⟩ = ⟨k, by { subst h, exact w }⟩ :=
+rfl
+
+@[simp] lemma fin_congr_symm_apply_mk {n m : ℕ} (h : n = m) (k : ℕ) (w : k < m) :
+  (fin_congr h).symm ⟨k, w⟩ = ⟨k, by { subst h, exact w }⟩ :=
+rfl
```

Comment:
```suggestion
@[simp] lemma fin_congr_symm_ {n m : ℕ} (h : n = m) (k : ℕ) (w : k < m) :
  (fin_congr h).symm = fin_congr h.symm := rfl
```
which makes  `fin_congr_symm_apply_coe` below redundant. Note this is essentially the same statement as `equiv.subtype_equiv_symm`